### PR TITLE
[WIP] - Feature/getting bt data

### PR DESCRIPTION
**Pull request by jacquesporveau**

### CODE REVIEW

File: db/migrate/20170310233802_create_solidus_paypal_braintree_transactions.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      t.string :nonce
+      t.string :payment_type
+      t.string :status
+      t.float :amount
+      t.datetime :date
+    end
+
+    add_index :solidus_paypal_braintree_transactions, :braintree_customer_id, unique: true, name: 'index_bt_transactions_on_bt_customer_id'
+    add_index :solidus_paypal_braintree_transactions, :braintree_transaction_id, unique: true, name: 'index_bt_transactions_on_bt_gateway_id'
+    add_index :solidus_paypal_braintree_transactions, :braintree_transaction_id, unique: true, name: 'index_bt_transactions_on_bt_transaction_id'
```
Comment (jacquesporveau) [inline db/migrate/20170310233802_create_solidus_paypal_braintree_transactions.rb]: Change this names.

File: spec/helpers/solidus_paypal_braintree/braintree_adapter_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require 'spec_helper'
+
+describe SolidusPaypalBraintree::BraintreeAdapter do
```
Comment (brendandeere) [inline spec/helpers/solidus_paypal_braintree/braintree_adapter_spec.rb]: What kind of spec is this

File: spec/helpers/solidus_paypal_braintree/braintree_adapter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  subject { described_class }
+
+  describe '.all_transactions', vcr: {
+    cassette_name: 'braintree_adapter/all_transactions',
+    match_requests_on: [:method, :uri]
+  } do
+    let(:all_transactions) { subject.all_transactions }
```
Comment (brendandeere) [inline spec/helpers/solidus_paypal_braintree/braintree_adapter_spec.rb]: This is the actual subject of your spec

File: spec/helpers/solidus_paypal_braintree/braintree_adapter_spec.rb [hunk 4]
```diff
@@ -12,4 +12,13 @@
       expect(all_transactions).to be_a(Braintree::ResourceCollection)
     end
   end
+
+  describe '.transaction_by_id', vcr: {
+    cassette_name: 'braintree_adapter/transaction_by_id'
+  } do
+    let(:transaction) { subject.transaction_by_id('ah3fg1f3') }
```
Comment (brendandeere) [inline spec/helpers/solidus_paypal_braintree/braintree_adapter_spec.rb]: This should be your test subject

File: spec/helpers/solidus_paypal_braintree/braintree_adapter_spec.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+require 'spec_helper'
+
+describe SolidusPaypalBraintree::BraintreeAdapter do
+  subject { described_class }
+
+  describe '.all_transactions', vcr: {
+    cassette_name: 'braintree_adapter/all_transactions',
```
Comment (brendandeere) [inline spec/helpers/solidus_paypal_braintree/braintree_adapter_spec.rb]: Where is this cassette? It should be committed

File: app/helpers/solidus_paypal_braintree/braintree_adapter.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+module SolidusPaypalBraintree
+  module BraintreeAdapter
```
Comment (brendandeere) [inline app/helpers/solidus_paypal_braintree/braintree_adapter.rb]: Is this module actually providing anything to the gem? or is it just wrapping BT gem methods with new names?

Comment (jacquesporveau) [inline app/helpers/solidus_paypal_braintree/braintree_adapter.rb]: Did I miss the mark on this one? The reason I wanted to wrap them was to add a level of consistency to our queries. There are numerous was to make a braintree query and I kind of wanted to adopt a standard in the way I was going to get data.

Also I think having them in a helper makes them more testable. If the queries weren't wrapped and just called in the controller I would have to check the instance variables... which I guess isn't the worst thing now that I type it out.

**Pull request closed without merging**